test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetSourcesTests.cs

```diff
@@ -0,0 +1,467 @@
+
+                // Act
+                // Set the working directory to C:\, otherwise,
+                // the test will change the nuget.config at the code repo's root directory
+                // And, will fail since global nuget.config is updated
+                var result = _fixture.RunDotnet(root, string.Join(" ", args), ignoreExitCode: true);
+
+                // Assert
+                Assert.True(result.ExitCode == 0);
+                // Ensure that no messages are shown with Verbosity as Quiet
+                Assert.Equal(string.Empty, result.Item2);
+                var settings = Settings.LoadDefaultSettings(null, null, null);
+                var packageSourcesSection = settings.GetSection("packageSources");
+                var sourceItem = packageSourcesSection?.GetFirstItemWithAttribute<SourceItem>("key", "test_source");
+
+                Assert.Equal("http://test_source", sourceItem.GetValueAsPath());
+            }
+        }
+
+
+        /// <summary>
+        /// Verify non-zero status code and proper messages
+        /// </summary>
+        /// <remarks>Checks invalid arguments message in stderr, check help message in stdout</remarks>
+        /// <param name="commandName">The nuget.exe command name to verify, without "nuget.exe" at the beginning</param>
+        public void TestCommandInvalidArguments(string command, int badCommandIndex)
+        {
+            // Act
+            var result = _fixture.RunDotnet(Directory.GetCurrentDirectory(), command, ignoreExitCode: true);
+
+            var commandSplit = command.Split(' ');
+
+            // Break the test if no proper command is found
+            if (commandSplit.Length < 1 || string.IsNullOrEmpty(commandSplit[0]))
+                Assert.True(false, "command not found");
+
+            // 0th - "nuget"
+            // 1st - "source"
+            // 2nd - action
+            // 3rd - nextParam
+            string badCommand = commandSplit[badCommandIndex];
+
+            //TODO: item2 in dotnet.exe, item3 in nuget.exe - why doesn't it work the same?
+            // Assert command
+            Assert.Contains("'" + badCommand + "'", result.Item2, StringComparison.InvariantCultureIgnoreCase);
+
+
+            // Assert invalid argument message
+            string invalidMessage;
+            if (badCommand.StartsWith("-"))
+            {
+                invalidMessage = "error: Unrecognized option";
+            }
+            else
+            {
+                invalidMessage = "error: Unrecognized command";
+            }
+
+            // Verify Exit code
+            VerifyResultFailure(result, invalidMessage);
+            // Verify traits of help message in stdout
+            Assert.Contains("Specify --help for a list of available options and commands.", result.Item2);
+        }
+
+        /// <summary>
+        /// Utility for asserting faulty executions of nuget.exe
```

Comment:
If you opt to keep the method:

```suggestion
        /// Utility for asserting faulty executions of dotnet.exe
```